packages/platform-sdk/test/integrationTests/discount-code/discount-code.test.ts

```diff
@@ -0,0 +1,120 @@
+import { randomUUID } from 'crypto'
+import { apiRoot } from '../test-utils'
+import {
+  CartDiscountResourceIdentifier,
+  DiscountCodeDraft,
+  ResourceIdentifier,
+} from '../../../src'
+import {
+  createCartDiscount,
+  deleteCartDiscount,
+} from '../cart-discount/cart-discount-fixture'
+import {
+  createDiscountCode,
+  createDiscountCodeDraft,
+  deleteDiscountCode,
+} from './discount-code-fixture'
+
+describe('testing discount code API calls', () => {
+  it('should create and delete a discount code by ID', async () => {
+    const cartDiscount = await createCartDiscount()
+
+    const cartDiscountResourceIdentifier: CartDiscountResourceIdentifier[] = [
+      {
+        typeId: 'cart-discount',
+        id: cartDiscount.body.id,
+      },
+    ]
+
+    const discountCodeDraft: DiscountCodeDraft = {
+      name: { en: 'test-discount-code-' + randomUUID() },
+      code: randomUUID(),
+      cartDiscounts: cartDiscountResourceIdentifier,
+      cartPredicate: 'country="DE"',
+      isActive: false,
+    }
+
+    const responseCreatedDiscountCode = await apiRoot
+      .discountCodes()
+      .post({ body: discountCodeDraft })
+      .execute()
+
+    expect(responseCreatedDiscountCode.statusCode).toEqual(201)
+    expect(responseCreatedDiscountCode.body).not.toBe(null)
+
+    const responseDiscountCodeDeleted = await apiRoot
+      .discountCodes()
+      .withId({ ID: responseCreatedDiscountCode.body.id })
+      .delete({
+        queryArgs: { version: responseCreatedDiscountCode.body.version },
+      })
+      .execute()
+    await deleteCartDiscount(cartDiscount)
+    expect(responseDiscountCodeDeleted.statusCode).toEqual(200)
+  })
+
+  it('should get a discount code by ID', async () => {
+    const cartDiscount = await createCartDiscount()
+    const discountCodeDraft = await createDiscountCodeDraft(cartDiscount)
+    const discountCode = await createDiscountCode(discountCodeDraft)
+    const getDiscountCode = await apiRoot
+      .discountCodes()
+      .withId({ ID: discountCode.body.id })
+      .get()
+      .execute()
+    expect(getDiscountCode).not.toBe(null)
+    expect(getDiscountCode.body.id).toEqual(discountCode.body.id)
+
+    await deleteDiscountCode(discountCode)
+    await deleteCartDiscount(cartDiscount)
+  })
+
+  it('should query a discount code', async () => {
+    const cartDiscount = await createCartDiscount()
```

Comment:
here is the same like previous comment: "Leaving these methods here to show the customer step-by-step how to create/delete the final object"